### PR TITLE
DOC: update the MultiIndex.swaplevel docstring

### PR DESCRIPTION
**Pull request by xrmx**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################### Docstring (pandas.MultiIndex.swaplevel)  ###################
################################################################################

Swap level i with level j.

Calling this method does not change the ordering of anything.

Parameters
----------
i : int, str
    First level of index to be swapped. Can pass level name as string.
    Type of parameters can be mixed.
j : int, str
    Second level of index to be swapped. Can pass level name as string.
    Type of parameters can be mixed.

Returns
-------
MultiIndex
    A newly allocated MultiIndex

.. versionchanged:: 0.18.1

   The indexes ``i`` and ``j`` are now optional, and default to
   the two innermost levels of the index.

See Also
--------
MultiIndex : A multi-level, or hierarchical, index object for
    pandas objects.

Examples
--------
>>> mi = pd.MultiIndex(levels=[['a', 'b'], ['bb', 'aa']],
...                    labels=[[0, 0, 1, 1], [0, 1, 0, 1]])
>>> mi
MultiIndex(levels=[['a', 'b'], ['bb', 'aa']],
   labels=[[0, 0, 1, 1], [0, 1, 0, 1]])
>>> mi.swaplevel()
MultiIndex(levels=[['bb', 'aa'], ['a', 'b']],
   labels=[[0, 1, 0, 1], [0, 0, 1, 1]])

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.MultiIndex.swaplevel" correct. :
```

### CODE REVIEW

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -1775,22 +1775,44 @@ def droplevel(self, level=0):
 
     def swaplevel(self, i=-2, j=-1):
         """
-        Swap level i with level j. Do not change the ordering of anything
+        Swap level i with level j.
+
+        Calling this method does not change the ordering of anything.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/multi.py]: (I know it was already there, but now we are changing it) The ordering of what? Maybe just the 'values' ?

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -1775,22 +1775,44 @@ def droplevel(self, level=0):
...
         Parameters
         ----------
-        i, j : int, string (can be mixed)
-            Level of index to be swapped. Can pass level name as string.
+        i : int, str
+            First level of index to be swapped. Can pass level name as string.
+            Type of parameters can be mixed.
+        j : int, str
+            Second level of index to be swapped. Can pass level name as string.
+            Type of parameters can be mixed.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/multi.py]: Can you also mention the default values here?

File: pandas/core/indexes/multi.py [hunk 3]
```diff
@@ -1775,22 +1775,44 @@ def droplevel(self, level=0):
...
+            pandas objects.
+
+        Examples
+        --------
+        >>> mi = pd.MultiIndex(levels=[['a', 'b'], ['bb', 'aa']],
+        ...                    labels=[[0, 0, 1, 1], [0, 1, 0, 1]])
+        >>> mi
+        MultiIndex(levels=[['a', 'b'], ['bb', 'aa']],
+           labels=[[0, 0, 1, 1], [0, 1, 0, 1]])
+        >>> mi.swaplevel()
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/multi.py]: I would use here actual values for i and j, so eg `swaplevel(0, 1)`

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR!

For the example, I was thinking: would it make sense to show an example with an actual dataframe with a MultiIndex? Then you would see more the typical usage.

File: pandas/core/indexes/multi.py [hunk 4]
```diff
@@ -1775,22 +1775,44 @@ def droplevel(self, level=0):
...
+            Type of parameters can be mixed.
+        j : int, str
+            Second level of index to be swapped. Can pass level name as string.
+            Type of parameters can be mixed.
 
         Returns
         -------
-        swapped : MultiIndex
+        MultiIndex
+            A newly allocated MultiIndex
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: A new MultiIndex

File: pandas/core/indexes/multi.py [hunk 5]
```diff
@@ -1775,22 +1775,44 @@ def droplevel(self, level=0):
...
 
         .. versionchanged:: 0.18.1
 
            The indexes ``i`` and ``j`` are now optional, and default to
            the two innermost levels of the index.
 
+        See Also
+        --------
+        MultiIndex : A multi-level, or hierarchical, index object for
+            pandas objects.
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: I wouldn't have MultiIndex, rather add Series.swaplevel and DataFrame.swaplevel

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/multi.py (hunk 1 above)
Comment (xrmx) [reply pandas/core/indexes/multi.py]: Done.

File: pandas/core/indexes/multi.py (hunk 2 above)
Comment (xrmx) [reply pandas/core/indexes/multi.py]: Done.

File: pandas/core/indexes/multi.py (hunk 3 above)
Comment (xrmx) [reply pandas/core/indexes/multi.py]: Done.

File: pandas/core/indexes/multi.py (hunk 4 above)
Comment (xrmx) [reply pandas/core/indexes/multi.py]: Done.

File: pandas/core/indexes/multi.py (hunk 5 above)
Comment (xrmx) [reply pandas/core/indexes/multi.py]: Done.

**Comment by xrmx** [comment]
Updated PR after review, thanks!

**Review by jreback — APPROVED** [review]

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @xrmx !

**Pull request merged**